lib/matplotlib/scale.py

```diff
@@ -94,6 +94,96 @@ def get_transform(self):
         return IdentityTransform()
 
 
+class FuncTransform(Transform):
+     """
+     A simple transform that takes and arbitrary function for the
+     forward and inverse transform.
+     """
+
+     input_dims = 1
+     output_dims = 1
+     is_separable = True
+     has_inverse = True
+
+     def __init__(self, forward, inverse):
+         """
+         Parameters
+         ----------
+
+         forward: callable
+             The forward function for the transform.  This function must have
+             an inverse and, for best behavior, be monotonic.
+             It must have the signature::
+
+                def forward(values: array-like) -> array-like
+
+         inverse: callable
+            The inverse of the forward function.  Signature as ``forward``.
```

Comment:
This paragraph is indented one character less than the one for forward.

Also the type annotations actually also use a space *before* the colon (see rst syntax for definition lists, and https://github.com/numpy/numpydoc/issues/78).

Likewise for the docstring below.